src/Components/TopicArticles.jsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react'
+import { fetchAllArticles } from '../api'
+import { useParams } from 'react-router-dom'
+import ArticleCard from './ArticleCard'
+import { Link } from 'react-router-dom'
+
+function TopicArticles() {
+    const { slug } = useParams()
+    const [topicArticles, setTopicArticles] = useState([])
+
+
+    useEffect(() => {
+        fetchAllArticles({
+            params: {
+                topic: slug
+            }
+        }).then((res) => {
+            setTopicArticles(res)
+        })
+    })
```

Comment:
you could have added a topic params to your articles page you already have rather than doing a fresh one for topics :)